sympy/integrals/trigonometry.py

```diff
@@ -79,8 +79,25 @@ def trigintegrate(f, x, conds='piecewise'):
 
         # take smallest n or m -- to choose simplest substitution
         if n_ and m_:
-            n_ = n_ and (n < m)  # NB: careful here, one of the
-            m_ = m_ and not (n < m)  # conditions *must* be true
+
+            # Make sure to choose the positive one
+            # otherwise an incorrect integral can occur.
+            if n < 0 and m > 0:
+                m_ = True
+                n_ = False
+            elif m < 0 and n > 0:
+                n_ = True
+                m_ = False
+            # Both are negative so choose the smallest n or m
+            # in absolute value for simplest substitution.
+            elif (n < 0 and m < 0):
+                n_ = n > m
+                m_ = not (n > m)
+
+            # Both n and m are odd and positive
+            else:
```

Comment:
For this case only, I don't think we have test cases. Also when this case is reached, this is handled the same way as it used to be,
```
>>> n = m = -1
>>> trigintegrate(sin(x)**n*cos(x)**m, x)
-log(sin(x)**2 - 1)/2 + log(sin(x))
```